src/test/java/FlightsReporterTests.groovy

```diff
@@ -0,0 +1,197 @@
+import java.time.LocalDateTime
+import java.time.LocalTime;
+//import FlightsReporter
+
+import static  org.junit.jupiter.api.Assertions.*;
+
+class FlightsReporterTests {
+    List<Flight> flights = []
+
+    @Test
+    void testFlightLengthInMinutes() {
+        Flight fl1 = new Flight(id: "SOFLON11", from: FlightsReporter.Airports.SOFIA, destination: FlightsReporter.Airports.LONDON,
+            departure: LocalDateTime.of(2021, 8, 3, 10, 30),
+                arrival: LocalDateTime.of(2021, 8, 3, 13, 15),
+                passengers: 100, capacity: FlightsReporter.CAPACITIES[3], runningLate: false)
+
+        assertEquals(FlightsReporter.flightLengthInMinutes(fl1), 165)
+
+    }
+
+    @Test
+    void testGetAllFlightsTo() {
+        FlightsReporter.printFlights(flights)
+        assert FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.AMSTERDAM)
+                == [flights[0], flights[1]]
+
+        assertEquals(FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.SOFIA),
+                Arrays.asList(flights[8], flights[9]))
+
+        assertEquals(FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.LONDON),
+                Arrays.asList(flights[2], flights[4], flights[7]))
+
+        assertNotEquals(FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.NEW_YORK),
+                Arrays.asList(flights[5], flights[6], flights[7]))
+    }
+
+    @Test
+    void testGetAllFlightsFrom() {
+        assert FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.SOFIA) == [flights[1], flights[2], flights[3]]
+
+        assertEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.NEW_YORK), [])
+
+        assertNotEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.BERLIN), [])
+        assertEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.PARIS),
+                [flights[0], flights[5], flights[8]])
+    }
+
+    @Test
+    void testPassengerAmount() {
+        def fl1 = FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.NEW_YORK)
+        assert FlightsReporter.passengerAmount(fl1) == 0
+
+        assert FlightsReporter.passengerAmount(flights) == 1430
+        assertEquals(FlightsReporter.passengerAmountTo(flights, FlightsReporter.Airports.NEW_YORK), 360)
+        assertEquals(FlightsReporter.passengerAmountTo(flights, FlightsReporter.Airports.SOFIA), 230)
+        assertEquals(FlightsReporter.passengerAmountFrom(flights, FlightsReporter.Airports.SOFIA), 460)
+        assertEquals(FlightsReporter.passengerAmountFrom(flights, FlightsReporter.Airports.NEW_YORK), 0)
+    }
+
+    @Test
+    void testSortingByDepartureDate() {
+        def departureExpected = []
+        def fl1 = flights.collect()
+        departureExpected << flights[9] << flights[5] << flights[1] << flights[0] << flights[4] << flights[6]
+                << flights[2] << flights[3] << flights[8] << flights[7]
```

Comment:
Ugly. Please don't do that.

Reply:
Also why these flights? is that all ? Which is ignored?  

EDIT: it is about the order... but is it ASC or DESC ? What if you change the implementation or data, you'd have to rearrange 10 different indexes ... not easy to maintain